dev/com.ibm.ws.com.unboundid/src/com/ibm/ws/com/unboundid/InMemoryLDAPServer.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.com.unboundid;
+
+import java.io.InputStream;
+
+import com.unboundid.ldap.listener.InMemoryDirectoryServer;
+import com.unboundid.ldap.listener.InMemoryDirectoryServerConfig;
+import com.unboundid.ldap.listener.InMemoryListenerConfig;
+import com.unboundid.ldap.sdk.DeleteRequest;
+import com.unboundid.ldap.sdk.Entry;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.Modification;
+import com.unboundid.ldap.sdk.schema.Schema;
+
+/**
+ * An in memory UnboundID LDAP server. See ContextPoolTimeoutTest for an example on how to use this.
+ * Basic steps: Create an instance of this class passing in your base DNs, eg. o=ibm,c=us. Add users
+ * and groups via entries. Note you will need to build the LDAP tree, so if you want to create
+ * uid=user1,ou=users,o=ibm,c=us, then you need an entry for ou=users first. The WIM schema has
+ * been merged in, so WIM attributes are available when using wimInetOrgPerson object class.
+ * To simulate Active Directory, use simulatedMicrosoftSecurityPrincipal object class. This
+ * allows memberOf and samAccountName to be used. You can view the WIM schema in wimschema.ldif.
+ */
```

Comment:
I would maybe re-format this, though the content is good. You might want to mention the wimGroupOfNames class as well. Maybe a list of objectclasses a user would use with a short description of what they are?